react-hooks/exhaustive-deps warning in favorites package

## 🐛 Bug report

### Current Behavior

when running `npm run lint` the following warning occurs:

```sh
  13:9  warning  The 'onChangeHandler' function makes the dependencies of useMemo Hook (at line 51) change on every render. Move it inside the useMemo callback. Alternatively, wrap the 'onChangeHandler' definition into its own useCallback() Hook  react-hooks/exhaustive-deps
```

### Expected behavior

No warning occurs

### Reproducible example

Run `npm run lint` in root of project

### Suggested solution(s)

Use suggestion from warning

### Your environment

```bash
$ npx envinfo --npmPackages '@availity/*' --binaries

  Binaries:
    Node: 10.14.2 - ~/.nvm/versions/node/v10.14.2/bin/node
    Yarn: 1.17.3 - ~/.nvm/versions/node/v10.14.2/bin/yarn
    npm: 6.10.2 - ~/.nvm/versions/node/v10.14.2/bin/npm
  npmPackages:
    @availity/api-axios: ^5.4.0 => 5.4.1
    @availity/api-core: ^5.4.0 => 5.4.0
    @availity/localstorage-core: ^2.6.1 => 2.6.1
```

--- a/packages/favorites/FavoriteHeart.js
+++ b/packages/favorites/FavoriteHeart.js
@@ -10,16 +10,16 @@ const FavoriteHeart = ({ id, onChange, onMouseDown, ...props }) => {
   const [tooltipOpen, toggleTooltip] = useToggle(false);
   const [loading, toggleLoading] = useToggle(true);
 
-  const onChangeHandler = () => {
-    toggleFavorite();
+  const icon = useMemo(() => {
+    const onChangeHandler = () => {
+      toggleFavorite();
 
-    if (onChange) {
-      onChange(isFavorite);
-    }
-  };
+      if (onChange) {
+        onChange(isFavorite);
+      }
+    };
 
-  const icon = useMemo(
-    () => (
+    return (
       <span
         id={`av-favorite-heart-${id}`}
         role="button"
@@ -47,9 +47,8 @@ const FavoriteHeart = ({ id, onChange, onMouseDown, ...props }) => {
         <span className="icon default-filled" />
         <span className="icon selected-filled" />
       </span>
-    ),
-    [id, isFavorite, onChangeHandler, onMouseDown, props]
-  );
+    );
+  }, [id, isFavorite, onMouseDown, props, onChange, toggleFavorite]);
 
   useEffectAsync(async () => {
     toggleLoading(false);

